sphinx_gallery/gen_gallery.py

```diff
@@ -534,18 +534,23 @@ def get_subsections(srcdir, examples_dir, gallery_conf, check_for_index=True):
         subfolders = [
             subfolder
             for subfolder in subfolders
+            # Returns `None` if `index.rst` or gallery header found
```

Comment:
Right?
```suggestion
            # Returns `None` if `index.rst` or gallery header not found
```

Reply:
Turns out we are both wrong, this whole thing is pretty confusing.

https://github.com/sphinx-gallery/sphinx-gallery/blob/7ddaf923dcb542931e10907ab6e426191b9b48c2/sphinx_gallery/gen_rst.py#L433-L458

If there is an `index.rst` -> return None
If there is a gallery header -> return file path
If there is no `index.rst`, no gallery header and raise is false -> None

So we want a subsection dir to contain a gallery header, so we check that the return `is not None`, which only happens when there is a gallery header. Let me amend the comment to be what happens when return is not None, as this is what we are checking for anyway.

Reply:
Actually the `check_for_index` parameter name is pretty confusing here, what about we change it to `check_for_header` or `check_for_gallery_header` but thats a bit verbose.

Reply:
Either is fine by me!

Reply:
changed to `check_for_header`